students/david/src/01-fibonacci/__tests__/iterativeFibonacci.spec.ts

```diff
@@ -0,0 +1,69 @@
+import assert from 'node:assert';
+import { describe, it } from 'mocha';
+
+import iterativeFibonacci from '../iterativeFibonacci';
+
+describe('Given the function iterativeFibonacci', () => {
+  describe('When is called with a negative number', () => {
+    it('Then it should send `Number must be positive or equal to zero`', () => {
+      let consoleError:bigint;
+      const enteredNumber = -3;
+      const { message} = new assert.AssertionError({message:'Number must be positive or equal to zero'});
+      
+      try {
+      consoleError = iterativeFibonacci(enteredNumber);
+      }
+      catch (error){
+        assert.equal(error.message, message);
+      }
+    });
+  });
+
+  describe('When is called with a float number', () => {
+    it('Then it should send `Number must be an integer`', () => {
+      let consoleError:bigint;
+      const enteredNumber = 2.2;
+      const { message} = new assert.AssertionError({message:'Number must be an integer'});
+      
+      try {
+      consoleError = iterativeFibonacci(enteredNumber);
+      }
+      catch (error){
+        assert.equal(error.message, message);
+      }
+    });
+  });
+
+  describe('When is called with a 0', () => {
```

Comment:
this and next test cases are coupled to value definition. Maybe it should be renamed to somethin like:

`When is called with a base case, it should return its base value` or  the others `when is called with a small positive number, it should return it fib value` and another test case `when is called with a big positive number it should return its fib value`. 
With this definition we cover functional cases not value cases